test/unit/deferred.js

```diff
@@ -253,7 +306,48 @@ test( "jQuery.Deferred.then - deferred (progress)", function() {
 			return jQuery.Deferred(function( defer ) {
 				defer.resolve( a * b );
 			});
+		}),
+		done = assert.async();
+
+	piped.progress(function( result ) {
+		return jQuery.Deferred().resolve().then(function() {
+			return result;
+		}).then(function( result ) {
+			value3 = result;
 		});
+	});
+
+	defer.progress(function( a, b ) {
+		value1 = a;
+		value2 = b;
+	});
+
+	defer.notify( 2, 3 );
+
+	piped.then( null, null, function( result ) {
+		return jQuery.Deferred().resolve().then(function() {
+			return result;
+		}).then(function() {
+			assert.strictEqual( value1, 2, "first progress value ok" );
+			assert.strictEqual( value2, 3, "second progress value ok" );
+			assert.strictEqual( value3, 6, "result of filter ok" );
+			done();
+		});
+	});
+});
+
+test( "[DIVERGENT] jQuery.Deferred.pipe - deferred (progress)", function( assert ) {
+
+	assert.expect( 3 );
+
+	var value1, value2, value3,
+		defer = jQuery.Deferred(),
+		piped = defer.pipe( null, null, function( a, b ) {
+			return jQuery.Deferred(function( defer ) {
+				defer.resolve( a * b );
+			});
+		}),
+		done = assert.async();
 
 	piped.done(function( result ) {
```

Comment:
(first note that these are not the same `defer`)

Yes, it is perfectly intentional. Let's take an example.

Say you have a Deferred that can progress from 0 to 100 (%) and you have a series of actions happening when the progress reaches a certain point (say 50).

The naive approach is to use progress handlers and repeat something like this throughout your app:

```js
upload.progress( function( advancement ) {
    if ( advancement >=50 ) {
        action1();
    }
} );

// ... elsewhere in the code ...

upload.progress( function( advancement ) {
    if ( advancement >=50 ) {
        action2();
    }
} );

// etc
```

Another, less coupled approach is to create a Promise that is resolved when the first one reaches the threshold and to share this one instead:

```js
var uploadNearCompletion = upload.pipe( null, null, function( advancement ) {
    if ( advancement >= 50 ) {
        return jQuery.Deferred( function( defer ) {
            defer.resolve();
        } );
    }
} );

// ... elsewhere in the code ...

uploadNearCompletion.done( action1 );

// ... elsewhere in the code ...

uploadNearCompletion.done( action2 );

// etc
```

That way, when Bob-the-project-manager comes and tells you "near completion" is now 20%, it's just a matter of checking where the near-completion promise is created.